web3/contract.py

```diff
@@ -120,85 +118,36 @@ class Contract(object):
     src_map_runtime = None
     user_doc = None
 
-    def __init__(self,
-                 *args,
-                 **kwargs):
+    def __init__(self, address=None):
         """Create a new smart contract proxy object.
 
         :param address: Contract address as 0x hex string
         """
-        code = kwargs.pop('code', empty)
-        code_runtime = kwargs.pop('code_runtime', empty)
-        source = kwargs.pop('source', empty)
-        abi = kwargs.pop('abi', empty)
-        address = kwargs.pop('address', empty)
 
         if self.web3 is None:
             raise AttributeError(
                 'The `Contract` class has not been initialized.  Please use the '
                 '`web3.contract` interface to create your contract class.'
             )
 
-        arg_0, arg_1, arg_2, arg_3, arg_4 = tuple(itertools.chain(
-            args,
-            itertools.repeat(empty, 5),
-        ))[:5]
-
-        if is_list_like(arg_0):
-            if abi:
-                raise TypeError("The 'abi' argument was found twice")
-            abi = arg_0
-        elif is_address(arg_0):
-            if address:
-                raise TypeError("The 'address' argument was found twice")
-            address = arg_0
-
-        if arg_1 is not empty:
-            if address:
-                raise TypeError("The 'address' argument was found twice")
-            address = arg_1
-
-        if arg_2 is not empty:
-            if code:
-                raise TypeError("The 'code' argument was found twice")
-            code = arg_2
-
-        if arg_3 is not empty:
-            if code_runtime:
-                raise TypeError("The 'code_runtime' argument was found twice")
-            code_runtime = arg_3
-
-        if arg_4 is not empty:
-            if source:
-                raise TypeError("The 'source' argument was found twice")
-            source = arg_4
-
-        if any((abi, code, code_runtime, source)):
-            warnings.warn(DeprecationWarning(
-                "The arguments abi, code, code_runtime, and source have been "
-                "deprecated and will be removed from the Contract class "
-                "constructor in future releases.  Update your code to use the "
-                "Contract.factory method."
-            ))
-
-        if abi is not empty:
-            validate_abi(abi)
-            self.abi = abi
-        if code is not empty:
-            self.bytecode = code
-        if code_runtime is not empty:
-            self.bytecode_runtime = code_runtime
-        if source is not empty:
-            self._source = source
-
-        if address is not empty:
-            validate_address(address)
-            self.address = to_normalized_address(address)
+        if address:
+            self.address = self.normalize_property('address', address)
```

Comment:
Thoughts on doing this via a `property` with a `setter` which performs this normalization?  Could do the same with `abi`

Reply:
Runs into some issues with class property `abi` and pseudo-class property `address`. (the property setter confuses `@combomethod`s that have `self.address`)